### ISSUE TITLE
psmLabel.exe OutOfMemoryException with mgf file

### ISSUE THREAD

**Issue by yangkl96** [pFindStudio/pDeep3#3]
Hello,

I am trying to do fine tuning. In the psmLabel folder, I execute `./psmLabel.exe psmLabel.cfg`

My psmLabel.cfg looks like 

psm_type = none
mode = pDeep
num_psm_file = 1
psm_file1 = fineTune.txt
ms2_type = pf2
num_ms2_file = 1
ms2_file1 = fineTune.mgf
output_folder = 
mass_tol = 20
tol_type = ppm
NH3_loss = true
H2O_loss = true
Mod_loss = true
num_ion_type = 2
iontype1 = b|N_term|0
iontype2 = y|C_term|0
num_new_aa = 0

The beginning of my fineTune.txt looks like 
raw_name	scan	peptide	modinfo	charge	RTinSeconds
fineTune	52460	LTGIKHELQANCYEEVKDR	12,Carbamidomethyl[C];	3	2385.96
fineTune	55888	TLTSGGHAEHEGKPYCNHPCYAAMFGPK	16,Carbamidomethyl[C];20,Carbamidomethyl[C];	4	2531.79
fineTune	45512	AASAAGAAGSAGGSSGAAGAAGGGAGAGTRPGDGGTASAGAAGPGAATK		4	2090.23

One entry in my fineTune.mgf looks like this. I generated this mgf file with a custom script that extracts m/z and intensities from mzml.
BEGIN IONS
TITLE=NDKSEEEQSSSSVK||2
CHARGE=2
RTINSECONDS=857.49792
129.10250854492188	151798.046875
662.8162231445312	111920.2109375
663.3179321289062	69980.6953125
768.3424682617188	52116.39042314679
768.8406982421875	112151.742809255
769.341064453125	77266.18957932951
777.3516235351562	466720.8756115453
777.852783203125	342208.846936838
778.3536987304688	154017.14499325794
778.8551025390625	53635.14559882285
781.3970947265625	131074.828125
782.4034423828125	52187.515625
851.4120483398438	52498.75
980.4542236328125	59644.421875
1109.49755859375	72305.28125
1196.529296875	196234.140625
1197.53173828125	117642.29687499999
END IONS

I keep getting this error: `Unhandled Exception: OutOfMemoryException.`

How do I adapt my files to work for mgf instead of raw?

**Comment by jalew188** [comment]
Sorry, MGFReader has not been implemented yet, you can use pParse (http://pfind.ict.ac.cn/software/pParse/index.html) to convert mzML files into pf2 files (default option), then psmLabel can read them.

**Comment by yangkl96** [comment]
Thanks for the help, I've tried using "psmLabel.exe psmLabel.cfg", this time with a raw file.
My new psmLabel.cfg looks like
psm_type = none
mode = pDeep
num_psm_file = 1
psm_file1 = fineTune.txt
ms2_type = raw
num_ms2_file = 1
ms2_file1 = 23aug2017_hela_serum_timecourse_4mz_narrow_1.raw
output_folder =
mass_tol = 20
tol_type = ppm
NH3_loss = true
H2O_loss = true
Mod_loss = true
num_ion_type = 2
iontype1 = b|N_term|0
iontype2 = y|C_term|0
num_new_aa = 0

I am getting this error
Unhandled Exception: System.ArgumentException: The path is not of a legal form.
   at System.IO.Path.LegacyNormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.InternalGetDirectoryName(String path)
   at Engine.Label.SpecLabel._CreateDir(String folder)
   at Engine.Label.SpecLabel.MatchAllPSM_And_Save(List`1 psm_files, String folder)
   at pDeep_label.pLabel.Main(String[] args)

My raw file and cfg file are in the psmLabel folder. Do you know what the issue could be?

**Comment by jalew188** [comment]
It is a command line error, not psmLabel.exe's error.

**Comment by jalew188** [comment]
Oh I see, your output folder is empty in your cfg file

**Comment by jalew188** [comment]
@yangkl96 is problem solved?

**Comment by yangkl96** [comment]
Thank you, psmLabel.exe runs now. However, it produces an empty file, with only the header "spec	peptide	modinfo	mz	charge	b	b-NH3	b-H2O	b-ModLoss	y	y-NH3	y-H2O	y-ModLoss". I edited the psmLabel.cfg so that all files needed had their absolute paths provided

**Comment by jalew188** [comment]
Could you please share the cfg and fineTune.txt to me? And the narrow window DIA file as well.

**Comment by jalew188** [comment]
@yangkl96 For DIA data, you can just send me the PRIDE link.

**Comment by jalew188** [comment]
By the way, fine-tuning using DIA data may not obtain an accurate model, since fragments from different peptides may be interferred by each other.

**Comment by yangkl96** [comment]
[psmLabel.cfg.txt](https://github.com/pFindStudio/pDeep3/files/5449133/psmLabel.cfg.txt)
[fineTune.txt](https://github.com/pFindStudio/pDeep3/files/5449135/fineTune.txt)

The dataset is available at https://chorusproject.org/anonymous/download/experiment/32fa43c0f9ba486eb3eedeb689f87765
This will download all the human datasets from the study, and they don't provide a link to download an individual RAW file, so let me know if I can share with you directly the one RAW file

**Comment by jalew188** [comment]
@yangkl96 the ```raw_name``` column does not contain the correct raw name. You used ```fineTune``` in the column, but the raw should be ```23aug2017_hela_serum_timecourse_4mz_narrow_1```. So psmLabel.exe could not locate the correct raw file for your PSMs.

**Comment by yangkl96** [comment]
Thank you, that was a mistake from when I was trying to use fineTune.mgf for finetuning, and I didn't update the raw_name for the actual raw file. I updated fineTune.txt and it works successfully.

Now I am running tune_and_predict.get_prediction, and I am encountering a problem where the RT finetuning is successful, but the MS/MS finetuning fails. I am getting my scan numbers from the pepxml output of peptideProphet, if that helps. I can open a new issue if that would be better:
############################
test of pre-trained RT model
[pDeep Info] predicted 100 peptide precursors (RT) using 0.275 seconds
[pDeep Info] PCC of test RT = 0.979
[pDeep Info] testing RT time = 0.309s


C:\Users\kevin\anaconda3\envs\env\lib\site-packages\numpy\core\fromnumeric.py:3335: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\kevin\anaconda3\envs\env\lib\site-packages\numpy\core\_methods.py:161: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)

Epoch =   1, mean_cost = nan, time = 0.00s

Epoch =   2, mean_cost = nan, time = 0.00s

Epoch =   1, mean_cost = nan, time = 0.00s
Epoch =   2, mean_cost = nan, time = 0.00s
Model save as pDeep_tune.ckpt
[pDeep Info] tuning time = 3.165s



Epoch =   1, peplen =  21, Batch=   22, size =    6, cost = 0.02254, time = 0.03s
Epoch =   1, mean_cost = 0.02667, time = 0.97s
Epoch =   2, peplen =  42, Batch=   22, size =    1, cost = 0.05027, time = 0.04s
Epoch =   2, mean_cost = 0.04094, time = 0.70s
Epoch =   3, peplen =  14, Batch=   22, size =   13, cost = 0.02430, time = 0.03s
Epoch =   3, mean_cost = 0.02638, time = 0.66s
Epoch =   4, peplen =  22, Batch=   22, size =    4, cost = 0.04509, time = 0.03s
Epoch =   4, mean_cost = 0.02507, time = 0.66s

Epoch =   1, mean_cost = 0.02667, time = 0.97s
Epoch =   2, mean_cost = 0.04094, time = 0.70s
Epoch =   3, mean_cost = 0.02638, time = 0.66s
Epoch =   4, mean_cost = 0.02507, time = 0.66s
Model save as pDeep_RT_tune.ckpt
[pDeep Info] RT tuning time = 4.171s
